client/src/main/java/io/pravega/client/ClientConfig.java

```diff
@@ -230,6 +245,23 @@ public ClientConfig build() {
             if (controllerURI == null) {
                 controllerURI = URI.create("tcp://localhost:9090");
             }
+            String scheme = controllerURI.getScheme();
+            // If controllerURI is kind of <HOST>:<PORT> then will add TCP as default. 
+            // Otherwise if URI have some other Unsupported Scheme like http, https, ftp
+            // etc.
+            // in those cases will throw IllegalArgumentException.
+            if (!isValidScheme(scheme)) {
+                String uri = controllerURI.toString();
+                String[] uriParts = uri.split(":");
```

Comment:
This should be: 
`controllerURI.getHost()`
and later `controllerURI.getPort()`